diff --git a/scripts/emulator.sh b/scripts/emulator.sh
index e636aec3..4858d2d4 100755
--- a/scripts/emulator.sh
+++ b/scripts/emulator.sh
@@ -36,7 +36,7 @@ if [ -z "$FIRST_STAGE" ]; then
   export ASH_STANDALONE=1
   if [ $(./busybox id -u) -ne 0 ]; then
     # Re-exec script with root
-    exec /system/xbin/su 0 ./busybox sh $0
+    exec /system/xbin/su -c "nohup ./busybox sh $0&"
   else
     # Re-exec script with busybox
     exec ./busybox sh $0
@@ -59,7 +59,6 @@ done
 
 # Stop zygote (and previous setup if exists)
 magisk --stop 2>/dev/null
-stop
 if [ -d /dev/avd-magisk ]; then
   umount -l /dev/avd-magisk 2>/dev/null
   rm -rf /dev/avd-magisk 2>/dev/null
@@ -74,6 +73,9 @@ elif [ -f /sepolicy ]; then
 else
   ./magiskpolicy --live --magisk 2>&1
 fi
+echo -n "u:r:magisk:s0" > /proc/$$/attr/current
+setenforce 1
+stop
 
 MAGISKTMP=/sbin
